src/PhpSpec/Matcher/TriggerMatcher.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Matcher;
+
+use PhpSpec\Wrapper\Unwrapper;
+use PhpSpec\Wrapper\DelayedCall;
+use PhpSpec\Exception\Example\FailureException;
+use PhpSpec\Exception\Fracture\MethodNotFoundException;
+
+final class TriggerMatcher implements Matcher
+{
+    /**
+     * @var Unwrapper
+     */
+    private $unwrapper;
+
+    /**
+     * @param Unwrapper $unwrapper
+     */
+    public function __construct(Unwrapper $unwrapper)
+    {
+        $this->unwrapper = $unwrapper;
+    }
+
+    /**
+     * @param string $name
+     * @param mixed  $subject
+     * @param array  $arguments
+     *
+     * @return bool
+     */
+    public function supports($name, $subject, array $arguments)
+    {
+        return 'trigger' === $name;
+    }
+
+    /**
+     * @param string $name
+     * @param mixed  $subject
+     * @param array  $arguments
+     *
+     * @return DelayedCall
+     */
+    public function positiveMatch($name, $subject, array $arguments)
+    {
+        return $this->getDelayedCall(array($this, 'verifyPositive'), $subject, $arguments);
+    }
+
+    /**
+     * @param string $name
+     * @param mixed  $subject
+     * @param array  $arguments
+     *
+     * @return DelayedCall
+     */
+    public function negativeMatch($name, $subject, array $arguments)
+    {
+        return $this->getDelayedCall(array($this, 'verifyNegative'), $subject, $arguments);
+    }
+
+    /**
+     * @param callable    $callable
+     * @param array       $arguments
+     * @param int|null    $level
+     * @param string|null $message
+     *
+     * @throws \PhpSpec\Exception\Example\FailureException
+     */
+    public function verifyPositive($callable, array $arguments, $level = null, $message = null)
```

Comment:
You can typehint `callable` here - it's supported in all versions we support now (older code may be missing it)